ledger/src/transaction_address_lookup_table_scanner.rs

```diff
@@ -0,0 +1,64 @@
+use {
+    bincode::deserialize,
+    lazy_static::lazy_static,
+    solana_sdk::{
+        address_lookup_table::{self, instruction::ProgramInstruction},
+        pubkey::Pubkey,
+        sdk_ids::SDK_IDS,
+        transaction::SanitizedVersionedTransaction,
+    },
+    std::collections::HashSet,
+};
+
+lazy_static! {
+    static ref SDK_IDS_SET: HashSet<Pubkey> = SDK_IDS.iter().cloned().collect();
+}
+
+pub enum ScanResult {
```

Comment:
naming: `Result` is too generic to add any valuable context. also possibly incomplete vs (non)native naming abstraction is too much for the amount of relevant code, just adding cognitive load... also, the two variant shared the same type shape..

how about this?


```
pub struct ScannedExtension(s) {
    possibly_incomplete: bool,
    extended_keys: Vec<Pubkey>,
}
```